fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -134,12 +134,14 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
         final LocalDate idealDisbursementDate = loanApplicationTerms.getExpectedDisbursementDate();
 
         if (!scheduleParams.isPartialUpdate()) {
+            loanApplicationTerms.setOriginalProposedPrincipal(loanApplicationTerms.getPrincipal());
```

Comment:
Shouldn't originalProposedPrincipal be initialized in the LoanTerms constructor? At this point, the principal might have changed already, no?

Reply:
kk

Reply:
like no, but yha i will change it